File: 2206-detonate-the-maximum-bombs/detonate-the-maximum-bombs.cpp
class Solution {
public:
    typedef long long ll;
    int BFS(int node,int n,unordered_map<int,vector<int>>&adj){
        queue<int>q;
        vector<int>vis(n,false);
        q.push(node);
        vis[node]=true;
        int count = 0;
        while(!q.empty()){
            int curr = q.front();
            q.pop();
            count++;
            for(int &v:adj[curr]){
                if(!vis[v]){
                    q.push(v);
                    vis[v]=true;
                }
            }
        }
        return count;
    }
    int maximumDetonation(vector<vector<int>>& bombs) {
        int n = bombs.size();
        unordered_map<int,vector<int>>adj;
        for(int i=0;i<n;i++){
            ll xi = bombs[i][0];
            ll yi = bombs[i][1];
            ll ri = bombs[i][2];
            ll sqr = ri*ri;
            for(int j=0;j<n;j++){
                if (i==j) continue;
                ll xj = bombs[j][0];
                ll yj = bombs[j][1];

                if((((xi-xj)*(xi-xj))+((yi-yj)*(yi-yj)))<=sqr){
                    adj[i].push_back(j);
                }
            }
        }

        int maxBombs = 0;

        for(int i=0;i<n;i++){
            maxBombs = max(maxBombs,BFS(i,n,adj));
            if (maxBombs == n) break;
        }
        return maxBombs;
    }
};

File: 2206-detonate-the-maximum-bombs/detonate-the-maximum-bombs.py
import collections
class Solution:
    def bfs(self,start_node,adj):
        queue = collections.deque([start_node])
        visited = {start_node}
        while queue:
            curr = queue.popleft()
            for neighbour in adj[curr]:
                if neighbour not in visited:
                    visited.add(neighbour)
                    queue.append(neighbour)
        return len(visited)            
    def maximumDetonation(self, bombs: List[List[int]]) -> int:
        n = len(bombs)
        adj = collections.defaultdict(list)

        for i in range(n):
            for j in range(n):
                if i == j :
                    continue
                xi , yi , ri = bombs[i]
                xj , yj , rj = bombs[j]

                if (xi-xj)**2 + (yi-yj)**2 <= ri**2:
                    adj[i].append(j)

        max_bombs = 0
        for i in range(n):
            max_bombs = max(max_bombs,self.bfs(i,adj))

        return max_bombs            
        